src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets

```diff
@@ -16,6 +16,68 @@
   <Import Project="$(BuildNumberTarget)" Condition="Exists('$(BuildNumberTarget)')" />
 
   <!-- #################################### -->
+  <!-- Version Info -->
+  <!-- #################################### -->
+
+  <ItemGroup>
+    <VersionInfoFile Include="$(BinDir)VersionInfo.json"/>
```

Comment:
><VersionInfoFile Include="$(BinDir)VersionInfo.json"/> [](start = 4, length = 54)

One problem that this has is that if people  don't clean their bin\ dir often the version will be just wrong and outdated most of the time.